diff --git a/app/controllers/users_controller.rb b/app/controllers/users_controller.rb
index 7e18057..bfd8d0e 100644
--- a/app/controllers/users_controller.rb
+++ b/app/controllers/users_controller.rb
@@ -20,14 +20,16 @@ class UsersController < ApplicationController
   def participation
     render :status => :forbidden unless authenticated?
 
-    participated = Rails.cache.fetch 'user-' + session[:drupal_user_id].to_s + '-participated-in-' + @campaign.id.to_s do
+    account, participated = Rails.cache.fetch 'user-' + session[:drupal_user_id].to_s + '-participated-in-' + @campaign.id.to_s do
       user = User.find_by_uid(session[:drupal_user_id])
-      user.participated?(@campaign.id)
+      pc = user.participated?(@campaign.id)
+
+      [user, pc]
     end
 
-    unless participated
-      user.participations.create(intent: false, campaign_id: @campaign.id)
-      user.handle_mc(@campaign)
+    if !participated
+      account.participations.create(intent: false, campaign_id: @campaign.id)
+      account.handle_mc(@campaign)
     end
 
     # Bring them to the real campaign root path or source
diff --git a/app/models/campaign.rb b/app/models/campaign.rb
index cde5dbc..acbcf4d 100644
--- a/app/models/campaign.rb
+++ b/app/models/campaign.rb
@@ -28,6 +28,10 @@ class Campaign < ActiveRecord::Base
     end
   end
 
+  after_create do
+    Rails.cache.delete 'campaign-list'
+  end
+
   def gated? type
     self.gated == type
   end
diff --git a/db/schema.rb b/db/schema.rb
index 5149054..10597a1 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -38,7 +38,6 @@ ActiveRecord::Schema.define(:version => 20130816135750) do
     t.datetime "image_updated_at"
     t.string   "mailchimp"
     t.string   "mobile_commons"
-    t.string   "mailchimp_submit"
     t.string   "email_submit"
     t.string   "email_signup"
     t.string   "meme_header"
